tests/specs/builds/output-commonjs.ts

```diff
@@ -126,6 +126,31 @@ export default testSuite(({ describe }, nodePath: string) => {
 			expect(content).toMatch('exports.sayHello =');
 		});
 
+		test('{ type: commonjs, field: component, srcExt: mjs, distExt: cjs }', async () => {
+			await using fixture = await createFixture({
+				...packageFixture(),
+				'package.json': createPackageJson({
+					main: './dist/component.cjs',
+				}),
+				'tsconfig.json': createTsconfigJson({
+					compilerOptions: {
+						jsx: 'preserve',
```

Comment:
Can you change this to a different option and assert the output type too?


Currently, `preserve` doesn't actually seem to be preserving the JSX and needs to be passed in as a top-level option in esbuild as well:
https://esbuild.github.io/try/#dAAwLjIzLjEAeyBsb2FkZXI6ICdqc3gnLCBqc3g6J3ByZXNlcnZlJywgdHNjb25maWdSYXc6ICd7ICJjb21waWxlck9wdGlvbnMiOiB7ICJqc3giOiAicHJlc2VydmUiIH0gfScgfQBleHBvcnQgY29uc3QgYSA9IDxhPmhpPC9hPg

https://github.com/evanw/esbuild/commit/7a727e305e7cab9e13b8819aa87f1b0053729298

Reply:
When I set it to react or react-jsx, due to the absence of a type signature for react, a type error will be reported. Can I add `@types/react` for this?

Reply:
Sure go ahead, will review again tomorrow

Reply:
I have updated